src/Apps/ArtAdvisor/06-Discovery-Neartext/weaviate-db.ts

```diff
@@ -0,0 +1,264 @@
+import {
+  DiscoveryArtwork,
+  DiscoveryUser,
+  MongoID,
+  UUID,
+} from "Apps/ArtAdvisor/06-Discovery-Neartext/types"
+import _ from "lodash"
+import weaviate, { WeaviateClient, generateUuid5 } from "weaviate-ts-client"
+
+const DEFAULT_ARTWORKS_CLASS = "DiscoveryArtworks"
+const DEFAULT_USERS_CLASS = "DiscoveryUsers"
+
+export class WeaviateDB {
+  private client: WeaviateClient
+  private artworkClass: string
+  private userClass: string
+
+  constructor(
+    options: {
+      /** Weaviate host url */
+      url?: string
+      /** Name of artwork class to use.
+       * Be sure this class maps to the likes/dislikes of the
+       * user class that is being supplied here.
+       */
+      artworkClass?: string
+      /** Name of user class to use.
+       * Be sure this class's likes/dislikes are mapped to the
+       * same artwork class that is being supplied here.
+       */
+      userClass?: string
+    } = {}
+  ) {
+    const url = options.url || process.env.WEAVIATE_URL
+    if (!url) throw new Error("Please provide url or set WEAVIATE_URL")
+
+    this.client = weaviate.client({ host: url })
+    this.artworkClass = options.artworkClass || DEFAULT_ARTWORKS_CLASS
+    this.userClass = options.userClass || DEFAULT_USERS_CLASS
+  }
+
+  async getUsers() {
+    const response = await this.client.graphql
+      .get()
+      .withClassName(this.userClass)
+      .withFields("name _additional { id }")
+      .do()
+
+    const users = response.data.Get.DiscoveryUsers.map(user => {
+      return {
+        id: user._additional.id,
+        name: user.name,
+      }
+    })
+
+    return users
+  }
+
+  /**
+   * Fetch a user
+   */
+  async getUser({
+    internalID,
+    id,
+  }: {
+    /** Mongo ID for Gravity User */
+    internalID?: string
+    /** Weaviate UUID for DiscoveryUser */
+    id?: string
+  }): Promise<DiscoveryUser> {
+    if (!internalID && !id)
+      throw new Error("Must provide either internalID or id (UUID)")
+
+    if (internalID) throw new Error("Not implemented yet")
+
+    if (!id) {
+      // temporary, while using test users
+      throw new Error("Missing id")
+    }
+
+    const response = await this.client.graphql
+      .get()
+      .withClassName(this.userClass)
+      .withWhere({
+        path: ["id"],
+        operator: "Equal",
+        valueString: id,
+      })
+      .withFields(
+        `
+        name
+        likedArtworks {
+          ... on DiscoveryArtworks { internalID }
+        }
+        dislikedArtworks {
+          ... on DiscoveryArtworks { internalID }
+        }
+        `
+      )
+      .do()
+
+    const {
+      name,
+      likedArtworks,
+      dislikedArtworks,
+    } = response.data.Get.DiscoveryUsers[0]
+
+    return {
+      id,
+      // internalID, // TODO: once we create real Gravity users
+      name,
+      likedArtworkIds: (likedArtworks || []).map(w => w.internalID),
+      dislikedArtworkIds: (dislikedArtworks || []).map(w => w.internalID),
```

Comment:
To go back to our previous convo, once we standardize around creating real gravity users, we will default to setting an empty array on user creation and remove the need for the `|| []`, correct?

Reply:
Yeah I think so. There should be other simplifications throughout as well, knowing we can just pass the bson id to identify a user instead of the uuid.